library/std/src/io/impls.rs

```diff
@@ -565,6 +565,11 @@ impl<A: Allocator> Write for VecDeque<u8, A> {
         Ok(len)
     }
 
+    fn write_all_vectored(&mut self, bufs: &mut [IoSlice<'_>]) -> io::Result<()> {
+        self.write_vectored(bufs)?;
```

Comment:
Can you please write a comment above that says that this is OK because `write_vectored` for `VecDeque` always writes all data? It's not immediately obvious.

That said, I'm not convinced that this method here should be overriden. The default implementation will simply call `write_vectored` as well with some stuff around but I think it's perfectly fine to leave the default here. I don't think we win anything from overriding it.

Reply:
Or alternatively, move the body of `write_vectored` into `write_all_vectored` and call in the other direction.

Reply:
In https://github.com/rust-lang/rust/pull/137051, I inspected the codegen for default implementation of `write_all_vectored` for `io::Empty`, and it has a lot of slicing logic that can't be optimized out. The same would apply here. I think it's useful to specialize it.